enum.c

```diff
@@ -3848,122 +3848,130 @@ struct enum_sum_memo {
 };
 
 static void
-sum_iter(VALUE i, struct enum_sum_memo *memo)
+sum_iter_normalize_memo(struct enum_sum_memo *memo)
 {
-    const int unused = (assert(memo != NULL), 0);
-
-    long n = memo->n;
-    VALUE v = memo->v;
-    VALUE r = memo->r;
+    assert(FIXABLE(memo->n));
+    memo->v = rb_fix_plus(LONG2FIX(memo->n), memo->v);
```

Comment:
Is it intentional that `if (n != 0)` condition is removed?

Reply:
I was aware of that and thought it is okay (0+something must be something).  But on a second thought it might be a problem because `v` can be something like a String?

Reply:
Note: Strings cannot reach this path.